### PR TITLE
*: Support `alter table rename table`

### PR DESCRIPTION
**Pull request by zimulala**
See [URL](http://dev.mysql.com/doc/refman/5.7/en/alter-table.html)

### CODE REVIEW

File: plan/resolver.go [hunk 1]
```diff
@@ -154,6 +154,12 @@ func (nr *nameResolver) Enter(inNode ast.Node) (outNode ast.Node, skipChildren b
 		}
 	case *ast.AlterTableStmt:
 		nr.pushContext()
+		for _, spec := range v.Specs {
+			if spec.Tp == ast.AlterTableRenameTable {
+				nr.currentContext().inCreateOrDropTable = true
```
Comment (hanfei1991) [inline plan/resolver.go]: Why set this true ? What is the usage of this var?

Comment (zimulala) [inline plan/resolver.go]: It's the same with `ast.RenameTable`. The NewTableName doesn't exist.

File: ddl/table.go [hunk 2]
```diff
@@ -255,6 +255,7 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) error {
 		return errors.Trace(err)
 	}
 	job.State = model.JobDone
+	job.SchemaState = model.StatePublic
```
Comment (shenli) [inline ddl/table.go]: Is this a bug fix?

**Comment by shenli** [comment]
LGTM

**Pull request merged**